src/components/legend/draw.js

```diff
@@ -329,9 +329,31 @@ module.exports = function draw(gd) {
                 xf = dragElement.align(newX, 0, gs.l, gs.l + gs.w, opts.xanchor);
                 yf = dragElement.align(newY, 0, gs.t + gs.h, gs.t, opts.yanchor);
             },
-            doneFn: function(dragged) {
+            doneFn: function(dragged, numClicks, e) {
                 if(dragged && xf !== undefined && yf !== undefined) {
                     Plotly.relayout(gd, {'legend.x': xf, 'legend.y': yf});
+                } else {
+                    var traces = [],
+                        clickedTrace;
+                    fullLayout._infolayer.selectAll('g.traces').each(function() { d3.select(this).call(function() { traces.push(this); }); });
+                    for(var i = 0; i < traces.length; i++) {
+                        var tracei = traces[i];
+                        var p = tracei[0][0].getBoundingClientRect();
+                        if(e.clientX >= p.left && e.clientX <= p.right && e.clientY >= p.top && e.clientY <= p.bottom) {
+                            clickedTrace = tracei;
+                            break;
+                        }
+                    }
```

Comment:
I wrestled with this piece for a while. Your way definitely looks cleaner, but I think may need a slight adjustment since what comes back from the `selectAll` isn't the full object and would still need to be passed to `d3.select()` in order to get the right `g` that works within `handleClick()`